scripts/tools/zap/generate.py

```diff
@@ -87,10 +88,23 @@ def runPrettifier(templates_file, output_dir):
     try:
         jsonData = json.loads(Path(templates_file).read_text())
         outputs = [(os.path.join(output_dir, template['output'])) for template in jsonData['templates']]
-        outputs = list(filter(lambda filepath: os.path.splitext(filepath)[1] in listOfSupportedFileExtensions, outputs))
-        os.system('clang-format -i ' + ' '.join(outputs))
+        clangOutputs = list(filter(lambda filepath: os.path.splitext(filepath)[1] in listOfSupportedFileExtensions, outputs))
+        javaOutputs = list(filter(lambda filepath: os.path.splitext(filepath)[1] == ".java", outputs))
+
+        if len(javaOutputs) > 0:
+            # Keep this version in sync with what restyler uses (https://github.com/project-chip/connectedhomeip/blob/master/.restyled.yaml).
```

Comment:
First of all, we should have a comment there pointing to here.

Second, could we make use of `scripts/helpers/restyle-diff.sh` here instead of reimplementing things?  Or will its attempt to run things in a docker image mess with us if we are already running in a docker image?

Reply:
I've added a comment in `.restyled.yaml` pointing here.

I don't know if we should use `restyle-diff.sh`. I was mostly trying to keep the Java-specific code consistent with how we call `clang-format`. @vivien-apple Any comment here?

Reply:
I believe generate.py should have a post-step calling the restyler.

I believe restyler supports java and having a single automated restyler would be better than trying to keep several in sync.

@vivien-apple how are the current generated files restyled if at all?
Alternative is to tell restyler to never restyle generated files in .restyled.yaml

Reply:
Currently generate.py restyles the files it generates as part of the generation process, at least the ones with extensions in that whitelist.

Reply:
Closing this for now - we may use `restyle-diff` in the future, but I need this PR to unblock #7439.

Reply:
I usually run restyle-diff after the generate so I would support doing the same here. I can approve to unblock you.

Reply:
Thanks! Filed #7732 and assigned to myself.